be/src/exprs/vectorized/arithmetic_operation.h

```diff
@@ -55,7 +55,8 @@ bool check_fpe_of_min_div_by_minus_one(LType lhs, RType rhs) {
 template <typename Op, PrimitiveType Type, typename = guard::Guard, typename = guard::Guard>
 struct ArithmeticBinaryOperator {
     template <typename LType, typename RType, typename ResultType>
-    static inline ReturnType<Type, ResultType> apply(const LType& l, const RType& r) {
+    static inline ReturnType<Type, ResultType> apply(const LType& l, const RType& r, const ColumnPtr& l_column,
+                                                     const ColumnPtr& r_column) {
```

Comment:
This solution introduce some unnecessary parameters.

As my understanding, you need to some more information about the type, including precision and scale for decimal type. Indeed the Column contains this information, but other part is unnecessary.

Maybe, we could introduce the TypeInfo, which contains all information of a type?

Reply:
Yes, but we cannot use TypeInfo directly because it belongs to storage. And maybe better to introduce a context which contains all the extra information we need?